torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -4833,14 +4834,32 @@ std::tuple<Tensor, Tensor, Tensor> _trilinear_backward(
 }
 
 Tensor log1p_backward(const Tensor& grad, const Tensor& self) {
-  if (self.is_sparse()) {
-    AT_ERROR(
-        "log1p of a sparse tensor is made to be non-differentiable since ",
-        "local gradient of zero is 1 / (0 + 1) = 1 and it makes the tensor dense. ",
-        "Use a different mathematical operation which preserves sparsity of gradients, ",
-        "or report a bug if you think this is an error.");
-  }
-  return grad / (self + 1).conj();
+  // We must conditionally initalize this using to_dense if sparse, sparse
+  // addition is not supported without exact shape match
+  Tensor self_p1_conj;
+  if (self.layout() == c10::kSparse || self.layout() == c10::kSparseCsr ||
+      self.layout() == c10::kSparseCsc || self.layout() == c10::kSparseBsr ||
+      self.layout() == c10::kSparseBsc) {
+    // The warning only applies to the sparsity of self, dense grad is never
+    // materialized so if self is strided and grad is sparse nothing unepected
+    // happens memory wise
+    TORCH_WARN(
+        "log1p_backward: recived self with sparse layout, but backward requires materialization of a dense tensor with this shape");
+    self_p1_conj = (self.to_dense() + 1).conj();
```

Comment:
Is it possible to avoid the call to `to_dense` and just create another sparse tensors with the op applied to vals?

Reply:
There is no guarantee that the sparsity patterns match when both `grad` and `self` are sparse. There is no guarantee that they are both even sparse.